ns1/record_helper.go

```diff
@@ -0,0 +1,74 @@
+package ns1
+
+import (
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/mutexkv"
+	ns1 "gopkg.in/ns1/ns1-go.v2/rest"
+	"gopkg.in/ns1/ns1-go.v2/rest/model/dns"
+	"log"
+	"strconv"
+)
+
+type RecordMutexKV struct {
+	mutexkv *mutexkv.MutexKV
+	tracker map[string][]interface{}
+}
+
+func (m *RecordMutexKV) Lock(client *ns1.Client, i interface{}, record string, zone string) error {
```

Comment:
```go
// Lock creates a TXT record in the given zone indicating that the given record is being altered, so that parallel executions of this provider are synchronized. This is safe for use by multiple goroutines. A non-nill error is returned if the TXT record cannot be created.
```